pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetType describes the subnet type.
+type SubnetType string
+
+const (
+	SubnetTypePublic  SubnetType = "public"
+	SubnetTypePrivate SubnetType = "private"
+)
+
+// VPCNetworkConfigurationSpec defines the desired state of VPCNetworkConfiguration
+type VPCNetworkConfigurationSpec struct {
+	// Namespaces to be applied.
+	AppliedToNamespaces []string `json:"appliedToNamespaces,omitempty"`
+	// Tier0 gateway or Tier0 VRF path.
+	Tier0Gateway string `json:"tier0Gateway,omitempty"`
+	// Edge cluster path on which the networking elements will be created.
+	EdgeCluster string `json:"edgeCluster"`
+	// NSX-T Project of the Namespace associated with.
+	NSXTProject string `json:"nsxtProject"`
+	// NSX-T IP Block paths used to allocate public Subnets.
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=5
+	PublicIPV4Blocks []string `json:"publicIPV4Blocks"`
```

Comment:
Earlier you said it is called IPPools in NSX-T API? Anyway, please make the naming consistent with NSX-T.

Reply:
Checked with Shailesh and they will change to public_ipv4_block since they are actually IP Blocks. There is a under review patch https://gitreview.eng.vmware.com/c/nsx/+/462535/15/mp/policy/policy-api/src/main/resources/api_spec/root/nsx-policy/policy-vpc.yml.